### PR TITLE
allow future blocks in tests

### PR DESCRIPTION
**Pull request by winsvega**
metropolis transition test

### CODE REVIEW

**Comment by chfast** [comment]
Can you give any description what this is about?

File: libethashseal/genesis/eip158Test.cpp [hunk 1]
```diff
@@ -39,7 +39,8 @@ R"E(
 		"blockReward": "0x4563918244F40000",
 		"registrar" : "0xc6d9d2cd449a754c494264e1809c50e34d64562b",
 		"networkID" : "0x1",
-		"chainID": "0x01"
+		"chainID": "0x01",
+		"allowFutureBlocks" : "1"
```
Comment (chfast) [inline libethashseal/genesis/eip158Test.cpp]: So we have values like `0x1`, `0x01` and `1`. WTF? Why it cannot be `true` / `false`?

Comment (winsvega) [inline libethashseal/genesis/eip158Test.cpp]: because there is no specification for json file format.

File: libethereum/ChainParams.cpp [hunk 2]
```diff
@@ -66,8 +66,16 @@ ChainParams ChainParams::loadConfig(string const& _json, h256 const& _stateRoot)
 	cp.tieBreakingGas = params.count("tieBreakingGas") ? params["tieBreakingGas"].get_bool() : true;
 	cp.blockReward = u256(fromBigEndian<u256>(fromHex(params["blockReward"].get_str())));
 	for (auto i: params)
-		if (i.first != "accountStartNonce" && i.first != "maximumExtraDataSize" && i.first != "blockReward" && i.first != "tieBreakingGas")
+	{
+		if (i.first != "accountStartNonce" && i.first != "maximumExtraDataSize" && i.first != "blockReward" && i.first != "tieBreakingGas" && i.first != "allowFutureBlocks")
```
Comment (chfast) [inline libethereum/ChainParams.cpp]: This line does not fit on my FullHD display. Can you split it please?

File: test/tools/jsontests/BlockChainTests.cpp [hunk 3]
```diff
@@ -461,7 +461,6 @@ void overwriteBlockHeaderForTest(mObject const& _blObj, TestBlock& _block, Chain
 			BlockHeader parentHeader = importedBlocks.at(importedBlocks.size() - 1).blockHeader();
 			tmp.setTimestamp(toInt(ho["RelTimestamp"]) + parentHeader.timestamp());
 			tmp.setDifficulty(((const Ethash*)sealEngine)->calculateDifficulty(tmp, parentHeader));
-			this_thread::sleep_for(chrono::seconds((int)toInt(ho["RelTimestamp"])));
```
Comment (chfast) [inline test/tools/jsontests/BlockChainTests.cpp]: Was this the cause of the BC tests slowdown?

Comment (winsvega) [inline test/tools/jsontests/BlockChainTests.cpp]: when filling the tests this line actually waits this amounts of seconds so the forged block wont throw a FutureTime exception.

**Comment by chfast** [comment]
You ware suppose to wait for Windows build!

**Pull request merged**